barman/infofile.py

```diff
@@ -878,6 +878,34 @@ def get_child_backup_info(self, child_backup_id):
 
         return None
 
+    def walk_backups_tree(self):
+        """
+        Walk through all the children backups of the current incremental backup.
+
+        :return: a generator of LocalBackupInfo objects for each child backup.
+        """
+
+        if self.children_backup_ids:
+            for child_backup_id in self.children_backup_ids:
+                backup_info = LocalBackupInfo(
+                    self.server,
+                    backup_id=child_backup_id,
+                )
+                yield from backup_info.walk_backups_tree()
+        yield self
+
+    def walk_to_root(self):
+        """
+        Walk through all the parent backups of the current backup.
+
+        :return: a generator of LocalBackupInfo objects for each parent backup.
+        """
+
+        backup_info = self.get_parent_backup_info()
+        while backup_info:
+            yield backup_info
+            backup_info = backup_info.get_parent_backup_info()
```

Comment:
Shouldn't we `yield self` as well?
I mean, just so we make both methods "compatible" in terms of what is returned.
Also, it may make it easier to build the `pg_combinebackup` command later.

Reply:
depends on the behaviour we want... my rationale was:
"I already know the object that is `self`, so I start returning objects from the first parent."
I see nothing wrong in yielding self too... is just that in my mind were were more often going to ignore it.

Reply:
Either way works for me, actually.
I think we can keep it as it is, and later change if we think this is necessary.

Reply:
Agreed.